src/controllers/controlpickermenu.cpp

```diff
@@ -1479,14 +1479,23 @@ void ControlPickerMenu::addPlayerControl(const QString& control,
     }
 
     if (samplerControls) {
-        QMenu* samplerControlMenu = new QMenu(tr("Samplers"), controlMenu);
-        controlMenu->addMenu(samplerControlMenu);
+        QMenu* samplerControlMainMenu = addSubmenu(tr("Samplers"), controlMenu);
+        QMenu* samplerControlMenu = samplerControlMainMenu;
+        QMenu* samplerResetControlMainMenu = nullptr;
         QMenu* samplerResetControlMenu = nullptr;
         if (resetControlMenu) {
-            samplerResetControlMenu = new QMenu(tr("Samplers"), resetControlMenu);
-            resetControlMenu->addMenu(samplerResetControlMenu);
+            samplerResetControlMainMenu = addSubmenu(tr("Samplers"), resetControlMenu);
+            samplerResetControlMenu = samplerResetControlMainMenu;
         }
+        const int maxSamplersPerMenu = 16;
+        int counter = 1;
```

Comment:
Maybe 
```suggestion
        int samplerInMenu = 0;
```
I would also initialize it with zero, because else you exceed  maxSamplersPerMenu.

Reply:
Done.
I applied the same renaming and the fix from your other suggestion also to WTrackMenu. Force-pushed but the changes were small to not cause review trouble I hope.